cdap-api/src/main/java/co/cask/cdap/internal/api/DefaultDatasetConfigurer.java

```diff
@@ -50,21 +53,43 @@
   }
 
   public void addStreams(Map<String, StreamSpecification> streams) {
+    Set<String> duplicateStreams = findDuplicates(streams.keySet(), this.streams.keySet());
```

Comment:
Why is this needed? Will we not be able to deploy two `ETLBatch` pipelines reading from same stream?